src/lib/expansion-panel/expansion-panel.ts

```diff
@@ -0,0 +1,142 @@
+  private _panelSubscriptions: Subscription[] = [];
+  /** QueryList of all expansion panels in the expansion panel set. */
+  @ContentChildren(forwardRef(() => MdExpansionPanel)) private _panels: QueryList<MdExpansionPanel>;
+
+  /** Set up event subscriptions for all panels for when they open, closing the other panels. */
+  ngAfterContentInit() {
+    this._panels.forEach((panel: MdExpansionPanel, idx: number, panels: MdExpansionPanel[]) => {
+      let panelId = panel.getExpansionPanelId();
+      let otherPanels = panels.filter(otherPanel => otherPanel.getExpansionPanelId() !== panelId);
+      let panelSubscription = panel.open.asObservable().subscribe(() => {
+        if (!this.multi) {
+          otherPanels.forEach(otherPanel => otherPanel.expanded = false);
+        }
+      });
+      this._panelSubscriptions.push(panelSubscription);
+    });
+  }
+
+  /** Clean up panel subscriptions. */
+  ngOnDestroy() {
+    this._panelSubscriptions.forEach(subscription => subscription.unsubscribe());
+  }
+}
+
+@Component({
+  moduleId: module.id,
+  styleUrls: ['./expansion-panel.css'],
+  selector: 'md-expansion-panel, mat-expansion-panel',
+  templateUrl: './expansion-panel.html',
+  encapsulation: ViewEncapsulation.None,
+  host: {
+    '[class.mat-expansion-panel]': 'true',
+    '[class.expanded]': 'expanded',
+    '[attr.wide]': 'wideStyling',
+    '[@flatStyling]': 'booleanToString(!flatStyling && expanded)',
+    '[@wideStyling]': 'booleanToString(wideStyling && expanded)',
+  },
+  animations: [
+    trigger('bodyExpansion', [
+      state('false', style({height: '0px', 'padding-bottom': '0'})),
+      state('true', style({height: '*', 'padding-bottom': '16px'})),
+      transition('false <=> true', animate(EXPANSION_PANEL_ANIMATION_TIMING)),
+    ]),
+    trigger('flatStyling', [
+      state('false', style({'margin-top': '*', 'margin-bottom': '*'})),
+      state('true', style({'margin-top': '16px', 'margin-bottom': '16px'})),
+      transition('false <=> true', animate(EXPANSION_PANEL_ANIMATION_TIMING)),
+    ]),
+    trigger('wideStyling', [
+      state('false', style({'margin-left': '*', 'margin-right': '*'})),
+      state('true', style({'margin-left': '0px', 'margin-right': '0px'})),
+      transition('false <=> true', animate(EXPANSION_PANEL_ANIMATION_TIMING)),
+    ]),
+  ],
+})
+export class MdExpansionPanel {
+  /** Whether flat styling should be used on expansion. */
+  get flatStyling(): boolean {
+    return !this.panelSet || this.panelSet.style === 'flat';
+  }
+  /** Whether wide styling should be used on expansion. */
+  get wideStyling(): boolean {
+    return this.panelSet && this.panelSet.style === 'wide';
+  }
+  /** Whether the expansion panel is expanded. */
+  private _expanded: boolean;
+  @Input('expanded')
+  get expanded(): boolean {
+    return this._expanded;
+  }
+  set expanded(expanded: boolean) {
+    this._expanded = expanded;
+    this.expandChange.emit(expanded);
+    if (expanded) {
```

Comment:
You can shorten this one to `expanded ? this.open.emit() : this.close.emit()`. Also shouldn't the event be called `closed`?